src/sentry/integrations/metric_alerts.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+
+from datetime import timedelta
+
+from sentry.incidents.logic import get_incident_aggregates
+from sentry.incidents.models import IncidentStatus, IncidentTrigger
+from sentry.utils.assets import get_asset_url
+from sentry.utils.http import absolute_uri
+
+QUERY_AGGREGATION_DISPLAY = {
+    "count()": "events",
+    "count_unique(tags[sentry:user])": "users affected",
+}
+
+
+def incident_attatchment_info(incident, metric_value=None):
+    logo_url = absolute_uri(get_asset_url("sentry", "images/sentry-email-avatar.png"))
+    alert_rule = incident.alert_rule
+
+    incident_trigger = (
+        IncidentTrigger.objects.filter(incident=incident).order_by("-date_modified").first()
+    )
+    if incident_trigger:
+        alert_rule_trigger = incident_trigger.alert_rule_trigger
+        # TODO: If we're relying on this and expecting possible delays between a trigger fired and this function running,
+        # then this could actually be incorrect if they changed the trigger's time window in this time period. Should we store it?
+        start = incident_trigger.date_modified - timedelta(
+            seconds=alert_rule_trigger.alert_rule.snuba_query.time_window
+        )
+        end = incident_trigger.date_modified
+    else:
+        start, end = None, None
+
+    if incident.status == IncidentStatus.CLOSED.value:
+        status = "Resolved"
+    elif incident.status == IncidentStatus.WARNING.value:
+        status = "Warning"
+    elif incident.status == IncidentStatus.CRITICAL.value:
+        status = "Critical"
+
+    agg_text = QUERY_AGGREGATION_DISPLAY.get(
+        alert_rule.snuba_query.aggregate, alert_rule.snuba_query.aggregate
+    )
+    if metric_value is None:
+        metric_value = get_incident_aggregates(incident, start, end, use_alert_aggregate=True)[
+            "count"
+        ]
+    time_window = alert_rule.snuba_query.time_window / 60
+
+    text = "{} {} in the last {} minutes".format(metric_value, agg_text, time_window)
+
+    if alert_rule.snuba_query.query != "":
+        text = text + "\nFilter: {}".format(alert_rule.snuba_query.query)
+
+    ts = incident.date_started
+
+    title = u"{}: {}".format(status, alert_rule.name)
+
+    return title, text, logo_url, status, ts
```

Comment:
@NisanthanNanthakumar Returning a tuple of 5 different things seems kind of risky to me since you are basing it off the order. I think it would make more sense to just return an object with the keys corresponding to the variable names would be safer. That way, you can't possibly mix up the order of return values and if you use a key that doesn't exist, it's going to error out in our unit tests.